pull_request_template.md

```diff
@@ -8,13 +8,16 @@
 [Discuss the impact of your changes on the project. This might include effects on performance, new dependencies, or changes in behaviour.]
 
 ## Checklist
-- [ ] My code adheres to the coding and style guidelines of the project.
-- [ ] I have performed a self-review of my own code.
-- [ ] I have extended current unit tests suite.
-- [ ] I have made corresponding changes to the documentation.
-- [ ] My changes generate no new warnings or new Sonar issues
-- [ ] I have passed the PR linting stage
-- [ ] I contacted my SRE colleague to update the configuration (if any changes)
+- [ ] **[JIRA]** I have linked the JIRA ticket to the PR.
+- [ ] **[CODE]** My code adheres to the coding and style guidelines of the project.
+- [ ] **[REVIEW]** I have performed a self-review of my own code.
+- [ ] **[DOCS]** I have made corresponding changes to the documentation.
+- [ ] **[SONAR]** My changes generate no new warnings or new Sonar issues.

```

Comment:
It's just my opinion, but I would advise against having the build fail if Sonar checks fail because in some cases Sonar might report a false positive like detecting an URL used just in tests (http instead of https) which can be bypassed easily by marking it as such. In any event if this happens one would have to re-trigger a build and then have Sonar pass this time.

Instead of SONAR do you think having LINT instead be a good idea?

Reply:
It's a good point that should be discussed on a broader scope with all the EMs. At this point all I'm trying with this PR is to bring clarity to the reader which items are checked and which are not especially since the template is quite bulky. Please check out the template again and let me know what you think so I can merge it.

FYI: @MamadouSy , @catalin-mihai-p5